spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/DataAddress.java

```diff
@@ -81,28 +84,23 @@ public Builder type(String type) {
         }
 
         public Builder property(String key, String value) {
-            Objects.requireNonNull(value);
+            Objects.requireNonNull(key, "Property key null.");
             address.properties.put(key, value);
             return this;
         }
 
-        public Builder properties(Map<String, ?> properties) {
-            // ArtifactRequestMessageImpl#urifyObjects (line 176): this "feature" converts every string starting with
-            // 'http' to become a URI.
-            // Thus, sometimes the <String, String> map can contain ... you guessed it ... URIs :(
-            properties.forEach((k, v) -> address.properties.put(k, v.toString()));
+        public Builder properties(Map<String, String> properties) {
```

Comment:
has the problem, that the ArtifactRequestMessageImpl converts strings to URIs, been taken care of elsewhere?

Reply:
This was really only a problem in some tests, where assets were flattened and passed directly to the properties() method. I don't think that currently serves any purpose. In any other case using Map<String, String> has not been an issue.

The idea with this change is that we shouldn't blindly cast anything to String. It'd be cleaner to just let anything that would generate non-String properties just typecast them itself. Which currently no component does.

So my current solution would be to not solve it at all. Does that make sense?

Reply:
This didn't just occur in tests, also in actual running code. I'm fine with leaving it as it is, as long as the end-to-end tests work, which unfortunately has to be done by hand. 

We cannot not solve it if the problem persists.

Reply:
That's a valid concern. Is there a readme for those tests? I'm not sure where to look.

Reply:
would be great to document which tests need to be executed in CONTRIBUTION.md so that we don't have to remind anybody and can just refer to the docs, @paullatzelsperger 

One observation: the properties method does not show a consistent behaviour over all builders we have (sometimes it adds values, sometime it overrides the whole map, sometime it string-object but stored as string-string, sometimes string-object and stored as string-object) - so this change feels like a step to a more consistent behaviour.

Reply:
It would be even better to have those e2e tests run automatically on every gradle build, which is currently being worked on. It will be able to run e.g. a file transfer from withing a JUnit test. If we document this now (which is fine with me), we'll have to remove it again soon.

On the other topic: I agree, Builders are somewhat inconsis tent, but that will require a broader refactoring. I propose:
- have a single final field (as opposed to invoking the object's CTor in the `build()` method
- if there are extensible properties, we have an `addProperty()` which adds a prop, and a `properties()` which replaces the entire map
WDYT?

Reply:
OK, just ran several tests on my local installation, all went fine. So I guess the original problem doesn't exist anymore. Will approve the PR.